mm2src/mm2_main/src/lp_wallet/mnemonics_wasm_db.rs

```diff
@@ -0,0 +1,163 @@
+use crypto::EncryptedMnemonicData;
+use mm2_core::mm_ctx::MmArc;
+use mm2_core::DbNamespaceId;
+use mm2_db::indexed_db::{DbIdentifier, DbInstance, DbUpgrader, IndexedDb, IndexedDbBuilder, OnUpgradeError,
+                         OnUpgradeResult, TableSignature};
+use mm2_err_handle::prelude::*;
+use std::collections::HashMap;
+
+type WalletsDBResult<T> = Result<T, MmError<WalletsDBError>>;
+
+#[derive(Debug, Deserialize, Display, Serialize)]
+pub enum WalletsDBError {
+    #[display(fmt = "Error deserializing '{}': {}", field, error)]
+    DeserializationError {
+        field: String,
+        error: String,
+    },
+    #[display(fmt = "Error serializing '{}': {}", field, error)]
+    SerializationError {
+        field: String,
+        error: String,
+    },
+    Internal(String),
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct MnemonicsTable {
+    wallet_name: String,
+    encrypted_mnemonic: String,
+}
+
+impl TableSignature for MnemonicsTable {
+    fn table_name() -> &'static str { "mnemonics" }
+
+    fn on_upgrade_needed(upgrader: &DbUpgrader, mut old_version: u32, new_version: u32) -> OnUpgradeResult<()> {
+        while old_version < new_version {
+            match old_version {
+                0 => {
+                    // do nothing explicitly because the table should be created on upgrade
+                    // from version 1 to 2 in order to avoid breaking existing databases
+                },
```

Comment:
There is `const DB_VERSION: u32 = 1;`, so `on_upgrade_needed` will "do nothing" at this point and won't reach `1` case. For now, there should be handling of only 2 variants (pseudocode):
```
0 => create_table,
unsupported_version => return error
```

Reply:
Right, it's literally written in the code comment 🤦 
Will fix it

Reply:
Fixed